File: app-mvvm/src/main/java/com/xuie/imaginary/gui/gank/show/Level1Item.java
package xuk.imaginary.gui.gank.show;

import com.chad.library.adapter.base.entity.AbstractExpandableItem;
import com.chad.library.adapter.base.entity.MultiItemEntity;

/**
 * @author xuie
 * @date 17-8-15
 */
public class Level1Item extends AbstractExpandableItem<Level1Item> implements MultiItemEntity {
    private String articleName;
    private String articleUrl;
    private String author;
    private String imageUrl;

    public String getArticleName() {
        return articleName;
    }

    public void setArticleName(String articleName) {
        this.articleName = articleName;
    }

    public String getArticleUrl() {
        return articleUrl;
    }

    public void setArticleUrl(String articleUrl) {
        this.articleUrl = articleUrl;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    @Override
    public int getLevel() {
        return 1;
    }

    @Override
    public int getItemType() {
        return ExpandableItemAdapter.TYPE_LEVEL_1;
    }
}

File: app-mvp-kotlin/src/main/java/xuk/imaginary/gui/gank/show/Level1Item.kt
package xuk.imaginary.gui.gank.show

import com.chad.library.adapter.base.entity.AbstractExpandableItem
import com.chad.library.adapter.base.entity.MultiItemEntity

/**
 * Created by xuie on 17-8-15.
 */

class Level1Item : AbstractExpandableItem<Level1Item>(), MultiItemEntity {
  var articleName: String? = null
  var articleUrl: String? = null
  var author: String? = null
  var imageUrl: String? = null

  override fun getLevel(): Int {
    return 1
  }

  override fun getItemType(): Int {
    return ExpandableItemAdapter.TYPE_LEVEL_1
  }
}
